securedrop/tests/test_journalist_api2.py

```diff
@@ -304,8 +307,186 @@ def test_api2_metadata_validation_valid_requests(
+        response = app.post(
+            url_for("api2.data"),
+            json=request_with_events,
+            headers=get_api_headers(journalist_api_token),
+        )
+        for event in request_with_events["events"]:
+            event_id = event["id"]
+            assert response.json["events"][event_id] == results[event_id]
+
+
+# FIXME: This is
+# "app/server_tests/data/items/40e13a88-5409-4201-9495-d06c335e203f.gpg" via
+# "gpg --enarmor".  Should probably pull from test_files fixture via
+# download_reply().
+REPLY = """-----BEGIN PGP MESSAGE-----
+Comment: Use "gpg --dearmor" for unpacking
+
+wcFMAwEQfVJow2WPAQ/9FAbkuKbTAu4WHk+iKNrEz21R0QeMDdKxffuQlD/36Gek
+gDqa4O8Nvkw4MfvprRuPwiXG6Jvm9++hiy1sjIlN/obIb9zUz/CfzQIrzOAipaBn
+OdwIc32s4hMtnnLdUZJa2vMKWFMyMAUrye3u0l7BgdBoDNUfDpKKLtDRtWGp0Uly
+5JkWfgVgfSEwzHkGKZvHI3EBVCt53eIyrK8B/KZ7NdMDtgzQgWb04pdWONx1SwKU
+72kIAgH7B44Btgn1MVj6Ri9IB470YZSWweIM0yTvQ/2//BPje6dCuK24vVmpz5Xd
+7tcZyqZYtIifwz0p4sfdoXMQmxMiyrCGmY7hosRjbbRFFVvI7yQ/ujEsdLqDbGok
+Ukv4gChYFMLOxqfpwF6v29A3MCHO3vwDBqQcwToQkP5BJE89jfF3+Z+n9+ahC6yX
+Gi1gYX+X0/1S9Q2kB1q+Pyqst4CtSiu9n+WJ4CoJXA27fafkUIWjxcu0GIzA+Y40
+2UzNiA4CRzj0rD9jOhDwCmrVqA/eR/nXdYK6wYnL7swGTHzD2HRf5p6fE6TnFSdt
+8K79sTDiVnH4S3AAS9vnL9HIBqhn6sSa/uojCazb7ZVcWexWLt2Mcd5ZFOpZ32qB
+Qf3Rhw4VSTJDZ4cIEDs531Pf+HZlIZRDEC64jNtTBVBJf9nn51cSxJmi4T8o4pLB
+wUwDw+fEwKIgGyoBEACLYAx2OqvbkscWu6Fp/zMM43omBiiEMQRAs87ldE1sddwk
+UE7N4H0xJE3l4x6poavY1oScEy+DiSvk6CYInEcDzGf6MSoCCBQ2cGjctfR84bE+
+mJV7M7P41AgV8Xj+NsfCcirwTrd1zir06/D3qg5JacKpscJFYJXJg0fBCFkFqiIv
+/6X1jcX9YAitLS3cLw+uV3ZuwKFUqnXLaclhvCvCTpdM+MuGvcNep+QFUeJnm+WS
+jVQPko4RiCOpTgH+g2pw1oBjVZC2UX0Iqake3Bnge89REs+zIzQ2SA+RhjVA6jR1
+rCf0ZIWckYg/WDxe2Dn4PAFhWgsjm4MM5dIE5YHvHPV8x0rIoAZQ8SwAXzTc1B1y
+pLq5Iop2AaJSQ+SFaiuFsUfc391kfsnOShQn00jLqZ7+bhWUWUS/rCH8ePa7Hp8W
+44MFOh8D3EsNN+hzuGXklHdL/dt41xmaO0o97yzusd8MJQ4fV0LBHkJEg5f21g3G
+bDpt5Fed7BWpR20cWrwuGPL46/UfjMqHJoS56ZjuZyBuUwvoC7gQlPmyPlvSWdEb
+8rF1gA4pPWURTNmClaHvPoubBig53mtXTz9esQfYu4FGmfUeFRnhdIWnrcvG3gXs
+/jY8gBIQ6N+MjkRNHS6nwzcUukStsaSvBI2uL0iclGFOCAx7p6TNQPDqwrITi9LA
+GwFbjruOwUUTjAErWxcTkGFWsUDPQV9gxamt7Agacinql4UAjSr21imTfqXkC/Kf
+6bgOb8EfLunqRg+44Zgk0JluiXYh+ss7alf/dcqeF1AMq+vhJ40F6r940IOWJ/0W
+cpZG68fwzC2YXTg4kU+OQdm4xBIeqTcgwiAnfZKKZtUCt8+JmRAzvbrLGXPe80Xc
+Dl9xuZb+mEQlOnxD3mYf6htx5CNRdp//Rl3fgbGv3vZCmE+GQ7CvgHkf+7Evinno
++eozX+auvPwAyXG4npnwjwEJ0XaTKAJwRVH26Q==
+=Eozu
+-----END PGP MESSAGE-----
+"""
+
+
+def test_api2_reply_sent(
+    journalist_app,
+    journalist_api_token,
+    test_files,
+    test_journo,
+):
+    """Test processing of the "reply_sent" event."""
+    with journalist_app.test_client() as app:
+        source_uuid = test_files["source"].uuid
+        index = app.get(
+            url_for("api2.index"),
+            headers=get_api_headers(journalist_api_token),
+        )
+
+        assert index.status_code == 200
+        source_version = index.json["sources"][source_uuid]
+
+        reply = {
+            "journalist_uuid": test_journo["uuid"],
```

Comment:
talking to @micahflee, it might make sense instead of sending the `journalist_uuid` with the pending event for the server to populate the `journalist_uuid` for an event based on the auth token of the request? this would make it easier for us to create events in offline mode (when we don't have the journalist UUID in the client) 

separately, I think (afaict) the event data right now is just typed as `Any`. it might be helpful to have the types for event data specified in the Python code? or in some batch API documentation, just to have more certainty that client + server match

Reply:
Re: typing:  I hear you!  I'll typedef `Event.data` for `reply_sent` and `item_deleted` tomorrow, either here or in a follow-up pull request depending on @legoktm's review.

Reply:
Proposed in #7681.